cmd/tempo-vulture/main.go

```diff
@@ -137,27 +144,72 @@ func main() {
 				continue
 			}
 
-			r := newRand(seed)
-			hexID := fmt.Sprintf("%016x%016x", r.Int63(), r.Int63())
+			log := logger.With(
+				zap.String("org_id", tempoOrgID),
+			)
+
+			client := util.NewClient(tempoQueryURL, tempoOrgID, log)
 
 			// query the trace
-			metrics, err := queryTempoAndAnalyze(tempoQueryURL, seed, hexID)
+			queryMetrics, err := queryTrace(client, seed)
 			if err != nil {
 				metricErrorTotal.Inc()
+				log.Error("query for metrics failed",
+					zap.Error(err),
+				)
+			}
+			pushMetrics(queryMetrics)
+		}
+	}()
+
+	// Search
+	go func() {
+		for now := range tickerSearch.C {
+			_, seed := selectPastTimestamp(startTime, now, interval, tempoSearchRetentionDuration)
+
+			// Don't attempt to read on the first itteration if we can't reasonably
+			// expect the write loop to have fired yet.  Double the duration here to
+			// avoid a race.
+			if seed.Before(actualStartTime.Add(tempoWriteBackoffDuration).Add(tempoWriteBackoffDuration)) {
+				continue
+			}
+
+			// Don't attempt to read future traces.
+			if seed.After(now) {
+				continue
 			}
 
-			metricTracesInspected.Add(float64(metrics.requested))
-			metricTracesErrors.WithLabelValues("requestfailed").Add(float64(metrics.requestFailed))
-			metricTracesErrors.WithLabelValues("notfound").Add(float64(metrics.notFound))
-			metricTracesErrors.WithLabelValues("missingspans").Add(float64(metrics.missingSpans))
-			metricTracesErrors.WithLabelValues("incorrectresult").Add(float64(metrics.incorrectResult))
+			log := logger.With(
+				zap.String("org_id", tempoOrgID),
+			)
+
+			client := util.NewClient(tempoQueryURL, tempoOrgID, log)
+
+			// query a tag we expect the trace to be found within
+			searchMetrics, err := searchTag(client, seed)
+			if err != nil {
+				metricErrorTotal.Inc()
+				log.Error("search for metrics failed",
+					zap.Error(err),
+				)
+			}
+			pushMetrics(searchMetrics)
 		}
 	}()
 
 	http.Handle(prometheusPath, promhttp.Handler())
 	log.Fatal(http.ListenAndServe(prometheusListenAddress, nil))
 }
 
+func pushMetrics(metrics traceMetrics) {
+	metricTracesInspected.Add(float64(metrics.requested))
+	metricTracesErrors.WithLabelValues("incorrectresult").Add(float64(metrics.incorrectResult))
+	metricTracesErrors.WithLabelValues("missingspans").Add(float64(metrics.missingSpans))
+	metricTracesErrors.WithLabelValues("tracemissingfromtagsearch").Add(float64(metrics.traceMissingFromTagSearch))
```

Comment:
`tracemissingfromtagsearch` is quite long winded. we are definitely in personal preference territory here, but maybe we could do something like:

`notfound_byid`
`notfound_search`

Reply:
Shorter and more readable sounds good to me.